### PR TITLE
Fixed: PowerShell modules would fail to download if they were found i…

### PR DESCRIPTION
**Pull request by jlariono**
…n more than one registered PowerShellGet repository.

### CODE REVIEW

File: Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1 [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                $Version = [version]('{0}.{1}.0' -f $TempVersion.Major,$TempVersion.Minor)
+            }
+        }
+
+        $module = Find-Module -Name $Name -AllVersions | 
+                        Where-Object { $_.Version.ToString() -like $Version } | 
+                        Sort-Object -Property 'Version' -Descending | 
+                        Select-Object -First 1
+
+        if( -not $module )
```
Comment (splatteredbits) [inline Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1]: Move this check after the if/else blocks.

File: Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1 [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            Write-Error -Message ('Failed to find module {0} version {1} on the PowerShell Gallery. Either the {0} module does not exist, or it does but version {1} does not exist. Browse the PowerShell Gallery at https://www.powershellgallery.com/' -f $Name, $tempVersion)
+            return
+        }
+    }
+    else
+    {
+        $module = Find-Module -Name $Name -ErrorAction Ignore | Select-Object -First 1
+        if( -not $module )
```
Comment (splatteredbits) [inline Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1]: Delete this check. The other check that got moved will work.

File: Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1 [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            if( $TempVersion -and ($TempVersion.Build -lt 0) )
+            {
+                $Version = [version]('{0}.{1}.0' -f $TempVersion.Major,$TempVersion.Minor)
+            }
+        }
+
+        $module = Find-Module -Name $Name -AllVersions | 
+                        Where-Object { $_.Version.ToString() -like $Version } | 
+                        Sort-Object -Property 'Version' -Descending | 
+                        Select-Object -First 1
```
Comment (splatteredbits) [inline Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1]: Remove this line. The other `Find-Module` call could also return multiple modules. We can do this check in one place.

File: Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1 [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+        if( -not $module )
+        {
+            Write-Error -Message ('Failed to find module {0} version {1} on the PowerShell Gallery. Either the {0} module does not exist, or it does but version {1} does not exist. Browse the PowerShell Gallery at https://www.powershellgallery.com/' -f $Name, $tempVersion)
+            return
+        }
+    }
+    else
+    {
+        $module = Find-Module -Name $Name -ErrorAction Ignore | Select-Object -First 1
```
Comment (splatteredbits) [inline Whiskey/Functions/Resolve-WhiskeyPowerShellModule.ps1]: Remove the `Select-Object` stage of the pipeline. Move it to line 71:

```
return $module | Select-Object -First 1
```

**Comment by jlariono** [comment]
@splatteredbits Requested changes made. Ready for another review.

File: Test/Get-WhiskeyPowerShellModule.Tests.ps1 [hunk 5]
```diff
@@ -62,10 +71,12 @@ function ThenModuleShouldBeInstalled
         $modulePath | should -Exist
     }
 
-    it ('Should have installed version {0}' -f $AtVersion) {
-        Find-Module -Name 'Pester' -AllVersions | Where-Object { $_.Version -match $AtVersion } | should -not -BeNullOrEmpty
+    if ($AtVersion)
```
Comment (splatteredbits) [inline Test/Get-WhiskeyPowerShellModule.Tests.ps1]: This should be its own `ThenModuleInstalledAtVersion` function.

**Comment by jlariono** [comment]
@splatteredbits Made requested changes and additional cleanup/improvements to GetPowerShellModule task tests. Ready for re-review.

**Pull request merged**